### PR TITLE
Stamp created date

### PR DESCRIPTION
**Pull request by AdeHarrison**
# Motivation and Context
Case record requires timestamping

# What has changed
Timestamp property added to Case entity

# How to test?
1. mvn clean install

2. tests should pass with new local eu.gcr.io/census-rm-ci/rm/census-rm-case-processor image created

# Links
[Trello](https://trello.com/c/MysxkDnC/776-defect-case-processor-should-stamp-created-date-or-case-api-should-infer-it-from-events)

# Screenshots (if appropriate):

### CODE REVIEW

**Comment by AdeHarrison** [comment]
[DEFECT: Case processor should stamp created date or Case API should infer it from Events](https://trello.com/c/MysxkDnC/776-defect-case-processor-should-stamp-created-date-or-case-api-should-infer-it-from-events)

File: src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java [hunk 1]
```diff
@@ -76,6 +77,8 @@
 
   @Column private String actionPlanId;
 
+  @Column private Date createdDateTime;
```
Comment (LukeLoze1) [inline src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java]: use LocalDateTime

Comment (NickGrantONS) [reply src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java]: there was some discussion the other day about storing all dates in UTC, but I agree that LocalDateTime seems to be the modern way

File: src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java [hunk 2]
```diff
@@ -76,6 +77,9 @@
 
   @Column private String actionPlanId;
 
+  @Column(columnDefinition = "timestamp with time zone")
+  private Date createdDateTime;
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java]: This should be `OffsetDateTime`

File: src/main/java/uk/gov/ons/census/casesvc/model/entity/Event.java [hunk 3]
```diff
@@ -17,7 +17,8 @@
 
   @ManyToOne private UacQidLink uacQidLink;
 
-  @Column private LocalDateTime eventDate;
+  @Column(columnDefinition = "timestamp with time zone")
+  private LocalDateTime eventDate;
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/census/casesvc/model/entity/Event.java]: This should be `OffsetDateTime`

File: src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java (hunk 2 above)
Comment (NickGrantONS) [reply src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java]: also don't need the `(columnDefinition = "timestamp with time zone")` part

File: src/main/java/uk/gov/ons/census/casesvc/model/entity/Event.java (hunk 3 above)
Comment (NickGrantONS) [reply src/main/java/uk/gov/ons/census/casesvc/model/entity/Event.java]: also don't need the `(columnDefinition = "timestamp with time zone")` part

File: src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java (hunk 2 above)
Comment (AdeHarrison) [reply src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java]: how will postgres hold time zone without columnDefinition?

File: src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java [hunk 4]
```diff
@@ -82,6 +83,9 @@
 
   @Column private String actionPlanId;
 
+  @Column(columnDefinition = "timestamp with time zone")
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java]: remove `(columnDefinition = "timestamp with time zone")`

File: src/main/java/uk/gov/ons/census/casesvc/model/entity/Event.java [hunk 5]
```diff
@@ -17,7 +18,8 @@
 
   @ManyToOne private UacQidLink uacQidLink;
 
-  @Column private LocalDateTime eventDate;
+  @Column(columnDefinition = "timestamp with time zone")
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/census/casesvc/model/entity/Event.java]: remove `(columnDefinition = "timestamp with time zone")`

File: src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java (hunk 2 above)
Comment (NickGrantONS) [reply src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java]: OffsetDateTime implies time zone

Comment (AdeHarrison) [reply src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java]: done

File: src/main/java/uk/gov/ons/census/casesvc/model/entity/Event.java (hunk 3 above)
Comment (AdeHarrison) [reply src/main/java/uk/gov/ons/census/casesvc/model/entity/Event.java]: done

File: src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java (hunk 4 above)
Comment (AdeHarrison) [reply src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java]: done

File: src/main/java/uk/gov/ons/census/casesvc/model/entity/Event.java (hunk 5 above)
Comment (AdeHarrison) [reply src/main/java/uk/gov/ons/census/casesvc/model/entity/Event.java]: done

**Pull request merged**